File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.c
struct ListNode* createNode(int value) {
    struct ListNode* newNode = (struct ListNode*)malloc(sizeof(struct ListNode));
    newNode->val = value;
    newNode->next = NULL;
    return newNode;
}

// Function to merge nodes
struct ListNode* mergeNodes(struct ListNode* head) {
    struct ListNode dummy;
    struct ListNode* current = &dummy;
    current->next = NULL;

    struct ListNode* temp = head;
    int sum = 0;

    while (temp != NULL) {
        if (temp->val == 0) {
            if (sum != 0) {
                current->next = createNode(sum);
                current = current->next;
                sum = 0;
            }
        } else {
            sum += temp->val;
        }
        temp = temp->next;
    }

    return dummy.next;
}

File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeNodes(ListNode head) {
        ListNode dummy = new ListNode(0);
        ListNode current = dummy;

        ListNode temp = head;
        int sum = 0;

        while (temp != null) {
            if (temp.val == 0) {
                if (sum != 0) {
                    current.next = new ListNode(sum);
                    current = current.next;
                    sum = 0;
                }
            } else {
                sum += temp.val;
            }
            temp = temp.next;
        }

        return dummy.next;
    }
}